druid/src/widget/list.rs

```diff
@@ -34,6 +34,126 @@ use crate::{
+/// To include the [`im`] feature, update Cargo.toml
+/// ```ignore
+/// [dependencies]
+/// // before
+/// druid = "0.7.0"
+/// // after
+/// druid = { version = "0.7.0", features = ["im"] }
+/// ```
+///
+/// # Examples
+///
+/// ## Define the data
+///
+/// ```
+/// # use druid::{ Data, Lens };
+/// use druid::im::Vector;
+/// #[derive(Clone, Data, Lens)]
+/// struct AppState {
+///     list_data: Vector<String>,
+/// }
+/// ```
+/// ## Create initial State
+///
+/// ```
+/// # use druid::{ Data, Lens};
+/// # use druid::im::Vector;
+/// # #[derive(Clone, Data, Lens)]
+/// # struct AppState {
+/// #     list_data: Vector<String>,
+/// # }
+/// let initial_state = AppState {
+///     list_data: Vector::from(
+///         vec!(
+///             "one".into(),
+///             "two".into(),
+///             "three".into(),
+///             "four".into(),
+///         )),
+/// };
+/// ```
+/// ## Create widgets
+///
+/// ```
+/// # use druid::widget::{ Label, List };
+/// # use druid::{ Data, Lens, Widget, WidgetExt, Env};
+/// # use druid::im::Vector;
+/// # #[derive(Clone, Data, Lens)]
+/// # struct AppState {
+/// #     list_data: Vector<String>,
+/// # }
+/// fn root() -> impl Widget<AppState> {
+///     let list = List::new(list_item)
+///         .lens(AppState::list_data);
+///     list
+/// }
+///
+/// fn list_item() -> impl Widget<String> {
+///     let label = Label::new(|data: &String, _env: &Env|
+///         data.clone());
+///     label
+/// }
+/// ```
+/// ## Scrollable List
+/// Commonly a list should be able to scroll, the [`Scroll`] widget can be used to implement this
+/// feature.
+/// See docs for more info.
+///
+/// ```
+/// # use druid::widget::{ Label, List, Scroll };
+/// # use druid::{ Data, Lens, Widget, WidgetExt, Env};
+/// # use druid::im::Vector;
+/// # #[derive(Clone, Data, Lens)]
+/// # struct AppState {
+/// #     list_data: Vector<String>,
+/// # }
+/// # fn list_item() -> impl Widget<String> {
+/// #     let label = Label::new(|data: &String, _env: &Env|
+/// #         data.clone());
+/// #     label
+/// # }
+/// fn root() -> impl Widget<AppState> {
+///     let list = List::new(list_item)
+///         .lens(AppState::list_data);
+///     Scroll::new(list)
+/// }
+/// ```
+/// ## Complex widgets
+/// List can be used with any complex widgets.
+/// ```
+/// # use druid::widget::{ Label, List };
+/// # use druid::{ Data, Lens, Widget, WidgetExt, Env};
+/// # use druid::im::Vector;
+/// #[derive(Clone, Data, Lens)]
+/// struct AppState {
+///     list_data: Vector<InnerState>,
+/// }
+///
+/// #[derive(Clone, Data, Lens)]
+/// struct InnerState {
+///     name: String,
+/// }
+///
+/// fn root() -> impl Widget<AppState> {
+///     let list = List::new(list_item)
+///         .lens(AppState::list_data);
+///     list
+/// }
+///
+/// fn list_item() -> impl Widget<InnerState> {
+///     let label = Label::new(|data: &InnerState, _env: &Env|
+///         data.name.clone());
+///     label
+/// }
+/// ```
```

Comment:
I think you should remove that part, or at least make it way shorter. That fact that widgets can be compose doesn't need to be documented in every container widget.